data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -25,13 +26,16 @@
 import io.micronaut.data.annotation.DataTransformer;
 import io.micronaut.data.annotation.IgnoreWhere;
 import io.micronaut.data.annotation.Join;
+import io.micronaut.data.annotation.JsonView;
```

Comment:
Query builder should decide if there is a JSON column by the `QueryResult` annotation, not by the `JsonView`

Reply:
It's not always JSON result. For example this method 
```
@JdbcRepository(dialect = Dialect.ORACLE)
public interface StudentViewRepository extends PageableRepository<StudentView, Long> {
  Double findMaxAverageGrade();
}
```
does not return JSON and QueryResult is not needed here, but builder needs to know it is building query for Json View which would look like this 
```
SELECT MAX(sv.DATA.averageGrade.number()) FROM "STUDENT_VIEW" sv
```
so there has to be some information for builder in the entity telling whether it is Json View or not.

Reply:
I don't like the idea of having some annotation on the entity in the first place and for the repository to check it.
So I would definitely remove the view flag on the persistent entity and check it from the repository method annotations.

My suggestion would be something like this:
```
@EntityRepresentation(TABULAR) // Default  
@EntityRepresentation(COLUMN) // The JSON DATA column case  
@EntityRepresentation(type = COLUMN, columnName = "CUSTOM_COLUMN_XYZ", type = "JSON") // Different options  
@JdbcRepository(dialect = Dialect.ORACLE)
public interface StudentViewRepository extends PageableRepository<StudentView, Long> {
  Double findMaxAverageGrade();
}
```
The `@JsonView` should be just an alias for `@EntityRepresentation(COLUMN)`.

The repository visitor can annotate the repository with`EntityRepresentation` if the annotation is present on the entity or on the repository's generic type argument.

The query builder will see the `EntityRepresentation` annotation from the annotation metadata and possibly annotate the method as `@QueryResult` for the runtime to understand the result

Reply:
Ok, I will try to change to look like you suggested. I thought I would need to annotate entity also for schema builder (that will implement later, when docker images are compatible etc.). I also need it for save/update and not just for query so it looked the simplest to me.
But, will try your suggestion, if I can figure out point in code where that can be done. Still, need to tell somewhere entity is of json view type.